### ISSUE TITLE
Ignoring errors with RacerD

### ISSUE THREAD

**Issue by mkr-plse** [facebook/infer#1289]
If there is a class from a library on which a thread safety violation is reported, and all instances of that error related to that class needs to be ignored, is there an appropriate annotation for that?

**Comment by ngorogiannis** [comment]
Yes, `@ThreadSafe(enableChecks = false)` should silence all warnings within a method/class.

**Comment by mkr-plse** [comment]
I don't want to disable all checks in a class.  However, ignore errors corresponding to the usage of a specific thread-unsafe class (from a third party jar for which the source is unavailable).

**Comment by ngorogiannis** [comment]
I see, there is the option `external-java-packages` (more convenient to use in an .inferconfig file, but also usable on command line), where any package prefix listed there will not be analysed (and therefore no reports will be issued for it).
For example, the field in .inferconfig would look like this

```
  "external-java-packages": [
    "com.google.",
    "java."
  ],
```

**Comment by mkr-plse** [comment]
I tried providing the package name on command line as well as in the `.inferconfig`. Neither works. 

Passing `--external-java-packages org.slf4j.Logger` to `infer` and see a defect containing: 
`Unprotected call to method void Logger.debug(String) of un-annotated interface org.slf4j.Logger`. 

My `.inferconfig` file has the following:

```
{
 "external-java-packages": [
    "org.slf4j.",
  ],
}
```

**Comment by bupt01** [comment]
Excuse me, may I ask if Infer is capable of detecting atomicity violations?

**Comment by ngorogiannis** [comment]
Can you say more about what you mean by atomicity? 

In general, Infer understands `volatile` with regards to data races, but only so as to silence data race reports.

**Comment by bupt01** [comment]
Thank you! I misunderstood infer earlier. I thought it could detect concurrency defects other than data races, such as atomicity violations. In fact, it cannot.
